packages/python/plotly/plotly/express/_core.py

```diff
@@ -1295,11 +1295,17 @@ def build_dataframe(args, constructor):
     # make copies of all the fields via dict() and list()
     for field in args:
         if field in array_attrables and args[field] is not None:
-            args[field] = (
-                dict(args[field])
-                if isinstance(args[field], dict)
-                else list(args[field])
-            )
+            if isinstance(args[field], dict):
+                args[field] = dict(args[field])
+            elif (
+                field in ["custom_data", "hover_data"]
+                and isinstance(args[field], str)
+                and "data_frame" in args.keys()
+                and args[field] in args["data_frame"].columns
+            ):
+                args[field] = [args[field]]
+            else:
+                list(args[field])
```

Comment:
this needs to be `args[field] = list(args[field])` otherwise we're not actually using the copy we're making!

Reply:
Ha good catch! I was failing tests on the main branch so didn't look too deep when I failed some on my branch 😅